src/middlewared/middlewared/plugins/auth.py

```diff
@@ -1,22 +1,62 @@
 import crypt
 from datetime import datetime, timedelta
-import pyotp
 import random
 import re
 import socket
 import string
 import time
 
+import psutil
+from psutil._common import addr
+import pyotp
+
 from middlewared.schema import accepts, Bool, Datetime, Dict, Int, Patch, returns, Str
 from middlewared.service import (
     ConfigService, Service, filterable, filterable_returns, filter_list, no_auth_required,
     pass_app, private, cli_private, CallError,
 )
 import middlewared.sqlalchemy as sa
-from middlewared.utils import run
 from middlewared.validators import Range
 
 
+def get_peer_process(remote_addr, remote_port):
+    for connection in psutil.net_connections(kind='tcp'):
+        if connection.laddr == addr(remote_addr, remote_port):
+            try:
+                return psutil.Process(connection.pid)
+            except psutil.ProcessNotFound:
+                return None
+
+
+def get_remote_addr_port(app):
+    remote_addr, remote_port = app.request.transport.get_extra_info("peername")
+    if remote_addr in ["127.0.0.1", "::1"]:
+        try:
+            x_real_remote_addr = app.request.headers["X-Real-Remote-Addr"]
+            x_real_remote_port = int(app.request.headers["X-Real-Remote-Port"])
+        except (KeyError, ValueError):
+            pass
+        else:
+            if process := get_peer_process(remote_addr, remote_port):
+                if process.name() == "nginx":
+                    try:
+                        with open("/var/run/nginx.pid") as f:
+                            nginx_pid = int(f.read().strip())
+                    except Exception:
```

Comment:
Actually, if getting `nginx_pid` or `ppid` raises an Exception, the check `if ppid == nginx_pid` will raise an exception because those 2 variables won't be defined.